src/transformers/models/imagebind/feature_extraction_imagebind.py

```diff
@@ -0,0 +1,413 @@
+    """
+    if (
+        isinstance(array, (list, tuple))
+        and isinstance(array[0], (list, tuple))
+        and isinstance(array[0][0], np.ndarray)
+    ):
+        if array[0][0].ndim == data_dim:
+            return [[base_array.astype(dtype=dtype) for base_array in clips] for clips in array]
+        else:
+            raise ValueError(
+                f"`For List[List[np.ndarray]]` inputs the internal `np.ndarray`s are expected to have dimension"
+                f" {data_dim} but got dimension {array[0][0].ndim}"
+            )
+    elif isinstance(array, (list, tuple)) and isinstance(array[0], np.ndarray):
+        if array[0].ndim == data_dim + 1:
+            return [[np.asarray(base_array, dtype=dtype) for base_array in clips] for clips in array]
+        elif array[0].ndim == data_dim:
+            return [[base_array.astype(dtype=dtype)] for base_array in array]
+        else:
+            raise ValueError(
+                f"For `List[np.ndarray]` inputs the internal `np.ndarray`s are expected to have dimension"
+                f" {data_dim} or {data_dim + 1} but got dimension {array[0].ndim}"
+            )
+    elif isinstance(array, np.ndarray):
+        array = array.astype(dtype=dtype)
+        if array.ndim == data_dim + 2:
+            return [list(clips) for clips in array]
+        elif array.ndim == data_dim + 1:
+            return [[clip] for clip in array]
+        elif array.ndim == data_dim:
+            return [[array]]
+        else:
+            raise ValueError(
+                f"`np.ndarray` inputs are expected to have dimension in"
+                f" `[{data_dim}, {data_dim + 1}, {data_dim + 2}]` but instead got {array.ndim}"
+            )
+    else:
+        raise ValueError(f"Could not make batched and clipped audio from {array}")
+
+
+# Adapted from https://github.com/facebookresearch/pytorchvideo/blob/main/pytorchvideo/data/clip_sampling.py#L346
+def uniform_chunk_sampling(
+    total_duration: float, chunk_duration: float, num_chunks: int
+) -> List[Tuple[Fraction, Fraction]]:
+    """
+    Uniformly sample `num_chunks` chunks of duration `chunk_duration` from an audio/video of total duration `total_duration`.
+
+    Args:
+        total_duration (`float`): s
+            Total duration of the audio/video.
+        chunk_duration (`float`):
+            Duration of each chunk.
+        num_chunks (`int`):
+            Number of chunks to sample.
+    """
+    chunk_duration_fraction = Fraction(chunk_duration)
+    max_possible_clip_start = Fraction(max(total_duration - chunk_duration, 0))
+    uniform_clip = Fraction(max_possible_clip_start / max(num_chunks - 1, 1))
+
+    result = []
+    for clip_index in range(num_chunks):
+        clip_start_sec = uniform_clip * clip_index
+        clip_end_sec = clip_start_sec + chunk_duration_fraction
+        result.append((clip_start_sec, clip_end_sec))
+
+    return result
+
+
+class ImageBindFeatureExtractor(SequenceFeatureExtractor):
```

Comment:
This looks good to me. However, there might be too many back and forth from numpy arrays to torch tensors.

IMO yo should do the operation numpy->torch.tensor and torch.tensor->numpy `only once`, i.e when `torchaudio` is available and you use `ta_kaldi.fbank`. 
Every other operations should be left in numpy array IMO. That way, you benefit from torchaudio fbank speedups while keeping the dependency to torch minimal.

Reply:
@ylacombe With reference to the above reply, the numpy->torch.tensor and torch.tensor->numpy is only done to create spectrogram just [here](https://github.com/huggingface/transformers/blob/f878996feac659a8e93701b4592e2329d54613f9/src/transformers/models/imagebind/feature_extraction_imagebind.py#L267) in the code file and is necessary because the operations ahead are done on the torch tensor because when `is_speech_available` is `True` `ta_kaldi.fbank()` will output a torch tensor as well. So, to facilitate that the spectogram created with numpy array has to be created into a torch tensor